File: web/Reusehive/src/model/itemDetail.ts
import BaseItem from "./baseItem.ts";

class ItemDetail {
    item: BaseItem;
    images: Array<String>;

    constructor(item?: BaseItem, images?: Array<String>) {
        this.item = item || new BaseItem();
        this.images = images || [];
    }
}

export default ItemDetail;

File: service/src/main/java/com/reusehive/entity/ItemDetail.java
package com.reusehive.entity;

import com.reusehive.entity.database.Item;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ItemDetail implements Serializable {
    private Item item;
    private List<String> images;
}
